sdks/python/apache_beam/ml/transforms/tft_transforms.py

```diff
@@ -0,0 +1,416 @@
+            output_column_name: str) -> Dict[str, common_types.TensorType]:
+    # TODO: Pending outputting artifact.
+    return {
+        output_column_name: tft.compute_and_apply_vocabulary(
+            x=data, **self._kwargs)
+    }
+
+  def __str__(self):
+    return "compute_and_apply_vocabulary"
+
+
+class Scale_To_ZScore(TFTOperation):
+  def __init__(
+      self,
+      columns: List[str],
+      *,
+      elementwise: bool = False,
+      name: Optional[str] = None,
+      **kwargs):
+    """
+    This function performs a scaling transformation on the specified columns of
+    the incoming data. It processes the input data such that it's normalized
+    to have a mean of 0 and a variance of 1. The transformation achieves this
+    by subtracting the mean from the input data and then dividing it by the
+    square root of the variance.
+
+    Args:
+      columns: A list of column names to apply the transformation on.
+      elementwise: If True, the transformation is applied elementwise.
+        Otherwise, the transformation is applied on the entire column.
+      name: A name for the operation (optional).
+
+    scale_to_z_score also outputs additional artifacts. The artifacts are
+    mean, which is the mean value in the column, and var, which is the
+    variance in the column. The artifacts are stored in the column
+    named with the suffix <original_col_name>_mean and <original_col_name>_var
+    respectively.
+    """
+    super().__init__(columns, **kwargs)
+    self.elementwise = elementwise
+    self.name = name
+
+  def apply(self, data: common_types.TensorType,
+            output_column_name: str) -> Dict[str, common_types.TensorType]:
+    artifacts = self.get_artifacts(data, output_column_name)
+    output = {output_column_name: tft.scale_to_z_score(x=data, **self._kwargs)}
+    output_dict = {output_column_name: output}
+    if artifacts is not None:
+      output_dict.update(artifacts)
+    return output_dict
+
+  def get_artifacts(self, data: common_types.TensorType,
+                    col_name: str) -> Dict[str, tf.Tensor]:
+    mean_var = tft.analyzers._mean_and_var(data)
+    shape = [tf.shape(data)[0], 1]
+    return {
+        col_name + '_mean': tf.broadcast_to(mean_var[0], shape),
+        col_name + '_var': tf.broadcast_to(mean_var[1], shape),
+    }
+
+  def __str__(self):
+    return "scale_to_z_score"
+
+
+class Scale_To_0_1(TFTOperation):
+  def __init__(
+      self,
+      columns: List[str],
+      elementwise: bool = False,
+      name: Optional[str] = None,
+      **kwargs):
+    """
+    This function applies a scaling transformation on the given columns
+    of incoming data. The transformation scales the input values to the
+    range [0, 1] by dividing each value by the maximum value in the
+    column.
+
+    Args:
+      columns: A list of column names to apply the transformation on.
+      elementwise: If True, the transformation is applied elementwise.
```

Comment:
What does "elementwise" mean?

Reply:
From TFT documentation on tft.scale_to_0_1

If true, scale each element of the tensor independently.

elementwise: IIUC, if true, it calculates scale_to_0_1 on entire dataset. if false, it calculates on individual tensors.

I am just passing it if any user might be interested.